commands/default/math.js

```diff
@@ -405,41 +573,103 @@ module.exports = function(env) {
 
             for (var row of this.data) {
                 var parsedRow = [];
-                for (var element of row) {
-                    var flattened = this.#unwrapAndFlattenTree(element);
-                    if (flattened.length == 1)
-                        flattened[0].unwrap();
-                    parsedRow.push(flattened.join(""));
-                }
+                for (var element of row)
+                    parsedRow.push(element.tokenize().join(""));
+
                 parsedComponents.push(parsedRow.join("&"));
             }
 
             return "\\begin{" + mmode + "}" + parsedComponents.join("\\\\") + "\\end{" + mmode + "}";
         }
 
+        /**
+         * Creates an empty array of the specified size
+         * @param {int} dimX the number of columns
+         * @param {int} dimY the number of rows
+         */
         empty(dimX, dimY) {
```

Comment:
An nxm matrix should have n rows and m columns, so it would make sense for the first argument to be the dimension in y direction